src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,102 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
+
+    @Test
+    public void register_validUser_registersSuccessfully() {
+        User user = createUser("user12345", "qwerty123", 20);
+
+        User registeredUser = registrationService.register(user);
+
+        Assertions.assertEquals(user, registeredUser);
+        Assertions.assertEquals(registeredUser, storageDao.get(registeredUser.getLogin()));
+    }
+
+    @Test
+    public void register_nullUser_throwsInvalidUserException() {
+        Assertions.assertThrows(InvalidUserException.class, () ->
+                registrationService.register(null));
```

Comment:
Throws out a notification that it is checked here so that in case of test failure they know what is wrong.This applies to everyone else.